### PR TITLE
Added Chapter 10

### PR DESCRIPTION
**Pull request by MarkWignall**
Added Chapter 10

### CODE REVIEW

File: chapter10/mdw/Ex2.pl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/perl
+use utf8;
+use strict;
+use warnings;
+
+use v5.12;
+use Benchmark qw(:all);
+
+chdir "/tmp/";  # the default is our home directory
```
Comment (a-panayotopoulos) [inline chapter10/mdw/Ex2.pl]: Comment no longer makes sense!

File: chapter10/mdw/Ex2.pl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+my @sorted;
+
+cmpthese(1000000, {
+
+	'old' => sub { @sorted = sort { -s $a <=> -s $b } glob '*'; },
+
+	'swartz' => sub { @sorted = map { $_->[0] } sort { $a->[1] <=> $b->[1] } map { [$_, -s $_] } glob '*'; },
+});
```
Comment (a-panayotopoulos) [inline chapter10/mdw/Ex2.pl]: Code looks good, but how did they compare?

File: chapter10/mdw/Ex4.pl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	print " with contents:\n";
+    } else {
+	print " an empty directory\n";
+    }
+    $tablevel += 1;
+    for my $name (@names) {
+      next if $name eq '.' or $name eq '..';
+      data_for_path("$path/$name");
+    }
+    $tablevel -= 1;
```
Comment (a-panayotopoulos) [inline chapter10/mdw/Ex4.pl]: This is a bit hacky, tbh. Surely passing `$tablevel` as a parameter to `$data_for_path` would be more in the spirit of recursive functions?

File: chapter10/mdw/Ex2.pl (hunk 1 above)
Comment (MarkWignall) [inline chapter10/mdw/Ex2.pl]: Lol, Opps.

File: chapter10/mdw/Ex2.pl (hunk 2 above)
Comment (MarkWignall) [inline chapter10/mdw/Ex2.pl]: I didn't run it on a big enough directory for there to be a significant difference.

File: chapter10/mdw/Ex4.pl (hunk 3 above)
Comment (MarkWignall) [inline chapter10/mdw/Ex4.pl]: Yeah, That might have been a more elegant way to do it.

**Pull request merged**